### ISSUE TITLE
Unhelpful error message when no data is available

### ISSUE THREAD

**Issue by milenaveneziani** [MPAS-Dev/MPAS-Analysis#202]
If one asks to compute climatologies over a 1-year time frame, and no data is available over that year, the resulting error message is very unhelpful:
```
/ccs/proj/cli115/pwolfram/anaconda_envs/default-2.7/lib/python2.7/site-packages/dask/array/rechunk.py:161: RuntimeWarning: divide by zero encountered in long_scalars
  div = shape[k] // v
/ccs/proj/cli115/pwolfram/anaconda_envs/default-2.7/lib/python2.7/site-packages/dask/array/rechunk.py:162: RuntimeWarning: divide by zero encountered in long_scalars
  mod = shape[k] % v
Traceback (most recent call last):
  File "./run_analysis.py", line 239, in <module>
    analysis(config)
  File "./run_analysis.py", line 165, in analysis
    moc_streamfunction(config)
  File "/autofs/nccs-svm1_home1/milena/MPAS-git-repositories/MPAS-Analysis/mpas_analysis/ocean/meridional_overturning_circulation.py", line 152, in moc_streamfunction
    variableMap, calendar)
  File "/autofs/nccs-svm1_home1/milena/MPAS-git-repositories/MPAS-Analysis/mpas_analysis/ocean/meridional_overturning_circulation.py", line 261, in _cache_velocity_climatologies
    chunking=chunking)
  File "/autofs/nccs-svm1_home1/milena/MPAS-git-repositories/MPAS-Analysis/mpas_analysis/shared/generalized_reader/generalized_reader.py", line 189, in open_multifile_dataset
    ds = mpas_xarray.process_chunking(ds, chunking)
  File "/autofs/nccs-svm1_home1/milena/MPAS-git-repositories/MPAS-Analysis/mpas_analysis/shared/mpas_xarray/mpas_xarray.py", line 574, in process_chunking
    ds = ds.chunk(chunks)
  File "/ccs/proj/cli115/pwolfram/anaconda_envs/default-2.7/lib/python2.7/site-packages/xarray/core/dataset.py", line 1070, in chunk
    for k, v in self.variables.items()])
  File "/ccs/proj/cli115/pwolfram/anaconda_envs/default-2.7/lib/python2.7/site-packages/xarray/core/dataset.py", line 1065, in maybe_chunk
    return var.chunk(chunks, name=name2, lock=lock)
  File "/ccs/proj/cli115/pwolfram/anaconda_envs/default-2.7/lib/python2.7/site-packages/xarray/core/variable.py", line 513, in chunk
    data = data.rechunk(chunks)
  File "/ccs/proj/cli115/pwolfram/anaconda_envs/default-2.7/lib/python2.7/site-packages/dask/array/core.py", line 1646, in rechunk
    return rechunk(self, chunks, threshold, block_size_limit)
  File "/ccs/proj/cli115/pwolfram/anaconda_envs/default-2.7/lib/python2.7/site-packages/dask/array/rechunk.py", line 224, in rechunk
    x = _compute_rechunk(x, c)
  File "/ccs/proj/cli115/pwolfram/anaconda_envs/default-2.7/lib/python2.7/site-packages/dask/array/rechunk.py", line 528, in _compute_rechunk
    assert new_idx == tuple(len(c) - 1 for c in chunks)
UnboundLocalError: local variable 'new_idx' referenced before assignment
```

**Comment by xylar** [comment]
Hopefully addressed by #203

**Comment by milenaveneziani** [comment]
Closing this for now as it should have been addressed by #203.
